test/mocks/genMocks.ts

```diff
@@ -0,0 +1,155 @@
+import "mocha";
+import {
+    BigNumber,
+    ContractTransaction,
+    ethers,
+    Signer,
+} from "ethers";
+import { Block, Provider } from "@ethersproject/providers";
+import { GenSale } from "../../src/contracts";
+
+export const feb23TsSeconds = 1677173138;
+export const feb24TsSeconds = 1677264314;
+
+export type GenMocks = {
+    signer: Partial<Signer>;
+    genSale: Partial<GenSale>;
+    currentSaleConfiguration: [
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber,
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber
+    ] & {
+        sellerWallet: string;
+        parentDomainId: BigNumber;
+        salePrice: BigNumber;
+        amountForSale: BigNumber;
+        mintlistMerkleRoot: string;
+        startingMetadataIndex: BigNumber;
+        folderGroupID: BigNumber;
+        transactionLimit: BigNumber;
+    };
+    currentSaleStarted: boolean;
+    currentAmountForSale: BigNumber;
+    currentTransactionLimit: BigNumber;
+    currentSalePrice: BigNumber;
+    currentDomainsSold: BigNumber;
+    currentSaleStartTimestamp: BigNumber;
+    currentLatestBlock: Partial<Block>;
+    currentSaleId: string;
+    currentSaleCounter: BigNumber;
+    currentPauseStatus: boolean;
+    currentSignerFunds: ethers.BigNumber;
+    currentDomainsPurchasedByAccount: BigNumber;
+    reset: () => void;
+};
+
+export function getDefaultMockGenSaleConfiguration():
+    [
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber,
+        string,
+        BigNumber,
+        BigNumber,
+        BigNumber
+    ] & {
+        sellerWallet: string;
+        parentDomainId: BigNumber;
+        salePrice: BigNumber;
+        amountForSale: BigNumber;
+        mintlistMerkleRoot: string;
+        startingMetadataIndex: BigNumber;
+        folderGroupID: BigNumber;
+        transactionLimit: BigNumber;
+    } {
+    return {
+        sellerWallet: "0x1234567890123456789012345678901234567890",
+        parentDomainId: BigNumber.from(123),
+        salePrice: ethers.utils.parseEther("1.0"), // 1 ^ 18th
+        mintlistSaleDuration: BigNumber.from(3600), // ~ 1hr (seconds)
+        amountForSale: BigNumber.from(50),
+        mintlistMerkleRoot: "0xabcdef1234567890abcdef1234567890abcdef12",
+        startingMetadataIndex: BigNumber.from(1),
+        folderGroupID: BigNumber.from(2),
+        transactionLimit: BigNumber.from(10),
+    } as any;
+}
+
+export let mocks: GenMocks = {
+    currentSaleStarted: false,
+    currentSalePrice: BigNumber.from(0),
```

Comment:
nit: prefixing every item with `current` seems redundant as everything will always be the `current` thing

Reply:
The values can change depending on the test, hence the `current`

Reply:
Not a huge deal, just naming, so I'm fine let it slide as it was just a nitpick anyways, but I think even if it changes its value it's still just the `salePrice` and `saleStarted` value, this adds "at a certain point in time` to the values which isn't necessary because the values are never held and compared later in the same test, and the `mocks` value resets between tests as well

Reply:
This honestly does not matter to me, Im inclined to keep it as "current" because thats what is in Ryan's pr that I branched off of